### ISSUE TITLE
Issue with layers from a remote service when the layer name contains multiple ':'

### ISSUE THREAD

**Issue by audetrobergem** [GeoNode/geonode#5512]
## Expected Behavior
Display the details for a layer from a remote service when the layer name contains multiple ':'.

## Actual Behavior
Clicking on a layer (with multiple ':' in its name) from a remote service throws a ValueError with this traceback:

File "/home/geonode/.virtualenvs/neec_geoportal/local/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)
File "/home/geonode/.virtualenvs/neec_geoportal/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)
File "/home/geonode/.virtualenvs/neec_geoportal/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)
File "/home/geonode/.virtualenvs/neec_geoportal/local/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/home/geonode/.virtualenvs/neec_geoportal/local/lib/python2.7/site-packages/geonode/layers/views.py" in layer_detail
  466.             ":") if ":" in layer.alternate else (None, layer.alternate)
Exception Type: ValueError at /layers/nhn:nhn:network:flowdirection_1
Exception Value: too many values to unpack

## Steps to Reproduce the Problem
  1. Register a remote service (https://maps.geogratis.gc.ca/wms/hydro_network_en)
  2. Select some layers that contains multiple ':' in their name (nhn:nhn:hydrography:waterbody_2 for example) and import resources
  3. Select the newly added layer to view its details.

## Specifications
  - GeoNode version: 2.10.1
  - Installation method: GeoNode Project on a virtual environment
  - Platform: Ubuntu 18.04
  - Additional details:

**Comment by gannebamm** [comment]
@audetrobergem 
I am unable to reproduce this issue. See:
[remote service](https://master.demo.geonode.org/services/2131/) and its [corresponding layer](https://master.demo.geonode.org/layers/httpsmapsgeogratisgccawmshydro_network_en:nhn:nhn:hydrography:waterbody_2)

**Comment by audetrobergem** [comment]
@gannebamm 
I tried again on my installation and it didn't worked. I also looked at the layers/views.py file and the master file is different from the file I have on my installation (2.10.1).

The line that causes the error on my installation is:
```python
":") if ":" in layer.alternate else (None, layer.alternate)
```

Here is the part of the views.py file on my installation that contains the line that cause the error [(link)](https://github.com/GeoNode/geonode/blob/2.10.1/geonode/layers/views.py#L495)
```python
if check_ogc_backend(geoserver.BACKEND_PACKAGE):
    from geonode.geoserver.views import get_capabilities
    workspace, layername = layer.alternate.split(
        ":") if ":" in layer.alternate else (None, layer.alternate)
```

And this is the same part on the master views.py [(link)](https://github.com/GeoNode/geonode/blob/master/geonode/layers/views.py#L512):
```python
if check_ogc_backend(geoserver.BACKEND_PACKAGE):
    if layer.has_time:
        from geonode.geoserver.views import get_capabilities
        workspace, layername = layer.alternate.split(
            ":") if ":" in layer.alternate else (None, layer.alternate)
```

I guess that the master demo site is using this file. Since the layers from the remote service doesn't have a temporal component, the line that caused the error is skipped. That solve the problem with this specific remote service but it would still be an issue if the remote service has temporal data. I don't have any example for now.

Cheers!

**Comment by gannebamm** [comment]
Ok. I am not sure how to resolve this. If the layer is time enabled the `:` are used as parameters, as far as I understand the code?
Can someone with more experience point me in the right direction?

**Comment by GeoTob** [comment]
The code simply splits a `<workspace>:<layername>` string into the two components. The issue could be solved by using `split(':', 1)` to split the string at the first occurrence of `:`, assuming that workspace names will never contain a colon while layer names can contains colons. This would work with GeoServer (GS disallows colons in workspace names), but I'm not sure about other backends.

What strikes me though is that neither `workspace` nor `layername` seem to be used anywhere later in the method and I'm wondering if we can simply delete the entire:

```
workspace, layername = layer.alternate.split(
            ":") if ":" in layer.alternate else (None, layer.alternate)
```

**Comment by gannebamm** [comment]
@GeoTob I thought the `:` is used to determine some kind of time parameters later, but I am unable to find anything related to it in the code I looked at yet. @afabiani Do you see any issues with the proposed deletion above?

**Comment by afabiani** [comment]
No objections about removing unused stuff and/or refuses.

**Comment by t-book** [comment]
I think @GeoTob is right: https://lgtm.com/projects/g/GeoNode/geonode/snapshot/4fa8a25d86113614eda29f0907f4f78cb2c563f7/files/geonode/layers/views.py?sort=name&dir=ASC&mode=heatmap#L515

**Comment by t-book** [comment]
@GeoTob  @gannebamm  as this is on progress at https://github.com/GeoNode/geonode/projects/4 any news here?

**Comment by gannebamm** [comment]
No news, yet.
@GeoTob Who will get this assigned? You, me?

**Comment by t-book** [comment]
@gannebamm 

> Who will get this assigned? You, me?

As you wish 😜

**Comment by gannebamm** [comment]
@GeoTob Is there any progress on your behalf? I was not able to do anything, yet.

**Comment by GeoTob** [comment]
@gannebamm we haven't yet started either, will let you know if that changes

**Comment by afabiani** [comment]
**WARNING: The fix is available on GeoNode 3.1 only**

Fixed through https://github.com/GeoNode/geonode/commit/04d105534dd8ed2ddd0a5a41bc827e1d41bcf32d

![image](https://user-images.githubusercontent.com/1278021/86572663-8616b200-bf73-11ea-9a59-25fe6e5a2229.png)
